nucliadb/src/nucliadb/common/external_index_providers/pinecone.py

```diff
@@ -84,23 +85,22 @@ def iter_matching_text_blocks(self) -> Iterator[TextBlockMatch]:
             vector_metadata = VectorMetadata.model_validate(matching_vector.metadata)  # noqa
             yield TextBlockMatch(
                 text=None,  # To be filled by the results hydrator
-                id=paragraph_id.full(),
-                resource_id=paragraph_id.field_id.rid,
-                field_id=paragraph_id.field_id.full(),
+                id=paragraph_id,
                 score=matching_vector.score,
                 order=order,
-                position_start=paragraph_id.paragraph_start,
-                position_end=paragraph_id.paragraph_end,
-                subfield_id=paragraph_id.field_id.subfield_id,
-                index=vector_id.index,
-                position_start_seconds=list(map(int, vector_metadata.position_start_seconds or [])),
-                position_end_seconds=list(map(int, vector_metadata.position_end_seconds or [])),
                 is_a_table=vector_metadata.is_a_table or False,
                 page_with_visual=vector_metadata.page_with_visual or False,
                 representation_file=vector_metadata.representation_file,
                 paragraph_labels=vector_metadata.paragraph_labels or [],
                 field_labels=vector_metadata.field_labels or [],
-                page_number=vector_metadata.page_number,
+                position=TextPosition(
+                    page_number=vector_metadata.page_number,
+                    index=vector_id.index or 0,
```

Comment:
Before it didin't have the `or 0`. Why is it being added? I guess it's a None, default to 0 now, but is that correct?

Reply:
It didn't have it there, but the only place where it was used was in the hydrate, where it computed the `TextPosition` and set `index=text_block.index or 0`, so I moved it here